core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -119,6 +125,10 @@ public synchronized ChannelImpl shutdown() {
       return this;
     }
     shutdown = true;
+    if (deadlineCancellationExecutor != null) {
+      deadlineCancellationExecutor =
```

Comment:
ah, thank you for the clarification, I should have investigated those guarantees more closely. :)